modules/installation-aws-upi-installation.adoc

```diff
@@ -0,0 +1,187 @@
+network settings. You must edit each of these MachineSets.
+
+. Edit each worker MachineSet to provide the correct values for your cluster:
++
+----
+$ oc edit machineset --namespace openshift-machine-api test-tkh7l-worker-us-east-2a -o yaml
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineSet
+metadata:
+  creationTimestamp: 2019-03-14T14:03:03Z
+  generation: 1
+  labels:
+    machine.openshift.io/cluster-api-cluster: test-tkh7l
+    machine.openshift.io/cluster-api-machine-role: worker
+    machine.openshift.io/cluster-api-machine-type: worker
+  name: test-tkh7l-worker-us-east-2a
+  namespace: openshift-machine-api
+  resourceVersion: "2350"
+  selfLink: /apis/machine.openshift.io/v1beta1/namespaces/openshift-machine-api/machinesets/test-tkh7l-worker-us-east-2a
+  uid: e2a6c8a6-4661-11e9-a9b0-0296069fd3a2
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      machine.openshift.io/cluster-api-cluster: test-tkh7l
+      machine.openshift.io/cluster-api-machineset: test-tkh7l-worker-us-east-2a
+  template:
+    metadata:
+      creationTimestamp: null
+      labels:
+        machine.openshift.io/cluster-api-cluster: test-tkh7l
+        machine.openshift.io/cluster-api-machine-role: worker
+        machine.openshift.io/cluster-api-machine-type: worker
+        machine.openshift.io/cluster-api-machineset: test-tkh7l-worker-us-east-2a
+    spec:
+      metadata:
+        creationTimestamp: null
+      providerSpec:
+        value:
+          ami:
+            id: ami-0eecbb884c8b35b1e <1>
+          apiVersion: awsproviderconfig.openshift.io/v1beta1
+          blockDevices:
+          - ebs:
+              iops: 0
+              volumeSize: 120
+              volumeType: gp2
+          credentialsSecret:
+            name: aws-cloud-credentials
+          deviceIndex: 0
+          iamInstanceProfile:
+            id: test-tkh7l-worker-profile
+          instanceType: m4.large
+          kind: AWSMachineProviderConfig
+          metadata:
+            creationTimestamp: null
+          placement:
+            availabilityZone: us-east-2a
+            region: us-east-2
+          publicIp: null
+          securityGroups: <2>
+          - filters:
+            - name: tag:Name
+              values:
+              - test-tkh7l-worker-sg
+          subnet: <3>
+            filters:
+            - name: tag:Name
+              values:
+              - test-tkh7l-private-us-east-2a
+          tags:
+          - name: kubernetes.io/cluster/test-tkh7l
+            value: owned
+          userDataSecret:
+            name: worker-user-data
+      versions:
+        kubelet: ""
+status:
+  fullyLabeledReplicas: 1
+  observedGeneration: 1
+  replicas: 1
+----
+<1> Specify the {op-system-first} AMI to use for your worker nodes.
+<2> Specify the name of the worker security group that you created.
+<3> Specify the subnet for the machines.
+
+.. Replace the `securityGroups` stanza with one that specifies the name of the
```

Comment:
@cuppett, do you have the format for the securityGroup and subnet stanzas that use names instead of tags?

Reply:
Here is one (lifted from a BZ):

```
      securityGroups:
      - filters:
        - name: tag:Name
          values:
          - worker-sg-jialiuuuu1
      subnet:
        filters:
        - name: tag:Name
          values:
          - jialiuuuu1-xbhm2-private-us-east-2b
```
There should also be a way to specify this using the always-there subnet or security group ID (instead of a tag of Name), but I'm not sure where these filters' formats are defined well. We should find them or a spot for them.

Reply:
The `subnet` entry is defined [here][1].  For UPI, it's probably easier to just set the subnet [ID or ARN][2].

[1]: https://github.com/openshift/cluster-api-provider-aws/blob/d19c6e58a9b2c5597a27f2f6ea4bd4a46f43c194/pkg/apis/awsproviderconfig/v1beta1/awsmachineproviderconfig_types.go#L130-L131
[2]: https://github.com/openshift/cluster-api-provider-aws/blob/d19c6e58a9b2c5597a27f2f6ea4bd4a46f43c194/pkg/apis/awsproviderconfig/v1beta1/awsmachineproviderconfig_types.go#L216-L226